### PR TITLE
workflows: Check documentation in `check_nims` and fix Sphinx errors/warnings

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Update `check_nims` workflow to generate API reference documentation and check for errors.

Upgrade to latest documentation tools.

Fix or suppress all Sphinx errors and warnings.

### Why should this Pull Request be merged?

Help developers avoid using invalid markup in docstrings.

### What testing has been done?

None.

### CODE REVIEW

**Review by mshafer-NI — COMMENTED** [review]
Looks good

File: ni_measurementlink_service/_datatypeinfo.py [hunk 1]
```diff
@@ -7,28 +7,23 @@
...
+    type_specialization: TypeSpecialization = TypeSpecialization.NoType
+    """Specific type when value_type can have more than one use."""
 
-        message_type (str): This is the gRPC full name of the message type.
-        Required when 'grpc_field_type' is Kind.TypeMessage.
-        Ignored for any other 'type'.
+    message_type: str = ""
+    """The gRPC full name of the message type. Required when 'grpc_field_type' is Kind.TypeMessage.
 
+    Ignored for any other 'type'.
```
Comment (dixonjoel) [inline ni_measurementlink_service/_datatypeinfo.py:24]: 'grpc_field_type'

**Review by dixonjoel — APPROVED** [review]

File: ni_measurementlink_service/measurement/service.py [hunk 2]
```diff
@@ -191,18 +189,16 @@ def convert_value_to_str(value: object) -> str:
     def register_measurement(self, measurement_function: _F) -> _F:
         """Register a function as the measurement function for a measurement service.
 
-        To declare a measurement function, use this idiom:
+        To declare a measurement function, use this idiom::
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply ni_measurementlink_service/measurement/service.py:192]: That's the rST syntax for a code block: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks

File: ni_measurementlink_service/_datatypeinfo.py [hunk 3]
```diff
@@ -7,28 +7,23 @@
...
+    """Field.Kind associated with the DataType."""
 
-        repeated: Whether the DataType is a repeated field
+    repeated: bool
+    """Whether the DataType is a repeated field."""
 
-        type_specialization: Specific type when value_type
-        can have more than one use
+    type_specialization: TypeSpecialization = TypeSpecialization.NoType
+    """Specific type when value_type can have more than one use."""
```
[the remark this answers is not in the document]
Comment (bkeryan) [reply ni_measurementlink_service/_datatypeinfo.py]: Updated

File: ni_measurementlink_service/_datatypeinfo.py (hunk 1 above)
Comment (bkeryan) [reply ni_measurementlink_service/_datatypeinfo.py]: Updated and converted to a real cross-reference.

(Then I noticed this file is internal, but whatever, Sphinx still creates docs for it.)

**Pull request merged**